.github/workflows/release-check.yml

```diff
@@ -2,134 +2,165 @@ name: Release Checker
 on: [ workflow_call ]
 
 jobs:
-  releaser:
+  prepare:
     runs-on: ubuntu-latest
     env:
-      TAG_EXISTS: "true"
-      VERSION: "" # the version number read from version.json
-      COMPARETO: "" # the version number to compare this version to
-      GORELEASE: ""
-      GOCOMPAT: ""
-      GOMODDIFF: ""
-      RELEASE_BRANCH_NOTE: ""
+      GITHUB_TOKEN: ${{ github.token }}
+    outputs:
+      version: ${{ steps.version.outputs.this }}
+      tag_exists: ${{ steps.tag_exists.outputs.this }}
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-go@v3
-        with:
-          go-version: "1.19.x"
-      - name: Determine version
-        env:
-          GITHUB_TOKEN: ${{ github.token }}
+      - id: version
+        name: Retrieve new version from version.json
         run: |
-          # If `version.json` file doesn't exists, `version` is `""` and `404` is printed on stderr.
-          # The step won't be marked as a failure though because the error happens in a subshell.
-          version="$(gh api -X GET 'repos/${{ github.event.pull_request.head.repo.full_name }}/contents/version.json' -f ref='${{ github.event.pull_request.head.sha }}' --jq '.content' | base64 -d | jq -r '.version')"
+          # https://docs.github.com/en/rest/reference/repos#get-repository-content
+          version="$(gh api -X GET 'repos/${{ github.event.pull_request.head.repo.full_name }}/contents/version.json' -f ref='${{ github.event.pull_request.head.sha }}' --jq '.content' | base64 -d | jq -r '.version // ""')"
+          status=$? && ([[ $status == 0 ]] || exit $status)
           echo "version=$version"
-          echo "VERSION=$version" >> $GITHUB_ENV
-      - name: Check if the tag already exists
-        # Check if a git tag for the version (as read from version.json) exists
-        # If that is the case, we don't need to run the rest of the workflow.
-        if: env.VERSION != ''
+          echo "::set-output name=this::$version"
```

Comment:
Should we use `echo "{name}={value}" >> $GITHUB_OUTPUT` ?

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/